File: 0242-valid-anagram/solution.c
bool isAnagram(char* s, char* t) {
    int m=strlen(s),n=strlen(t);
    if(m!=n) return false;
    int ans[26]={0};
    
    for(int i=0;i<m;i++){
        ans[s[i]-97]++;
        ans[t[i]-97]--;
    }
    for(int i=0;i<26;i++){
        if(ans[i]!=0)   return false;
    }
    return true;
}

File: 0242-valid-anagram/solution.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int n = s.length(), c = t.length();
        if(c!=n){
            return false;
        }
        HashMap<Character, Integer> m = new HashMap<>();

        for(char a : s.toCharArray()){
            m.put(a, m.getOrDefault(a,0)+1);
        }
        for(char b : t.toCharArray()){
            if(m.containsKey(b)){
                m.put(b, m.get(b)-1);
            }
            else{
                return false;
            }
        }
        for(char a :s.toCharArray()){
            if(m.get(a) != 0){
                return false;
            }
        }
        return true;
    }
}
